scala-fx/src/main/scala/fx/Continuation.scala

```diff
@@ -30,7 +30,7 @@ object Continuation:
 
   @tailrec def handleControl(control: Control[_], e: Throwable): Any =
     e match
-      case e: ExecutionException =>
+      case e: (ExecutionException | CompletionException) =>
```

Comment:
the error returned from the `Async.uncancelable` wraps a `ControlToken` in `CompletionException` and then this `ControlToken` is never matched in this pattern unless we add it here

Reply:
Does it exit now?

Reply:
Yeah. I see that. What I was calling "exiting" was getting into your tests and returning the test failure or success. We do need this fix here to unwrap the failure correctly to achieve that.

If we restructure it a bit to remove the `toEither` call in the fixture and return a `(Structured, Control[Throwable]` in the resource fixture, the test does "exit" we get results consistent with the behavior of the other `Async`-instance-using tests in other PRs -- a WrongThreadException.

More to come in the review.

Reply:
okay I see what you mean now. thank you for the suggestions 👍